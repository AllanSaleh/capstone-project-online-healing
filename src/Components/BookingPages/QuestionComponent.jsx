import React from 'react';

const QuestionComponent = ({ question, choices, setParentAnswer }) => {
  // const [selected, setSelected] = useState(false);

  const handleChoiceClick = (event) => {
    setParentAnswer(event.target.innerText);
  };
  return (
    <ul>
      {question && <h1 className="text-lg lg:text-subtitle">{question}</h1>}
      {choices &&
        choices.map((choice, index) => (
          <button
            key={index}
            type="button"
            onClick={handleChoiceClick}
            className="w-full my-4 p-4 lg:p-6 text-left text-md lg:text-paragraph hover:cursor-pointer bg-white border border-black border-opacity-20 rounded-lg"
          >
            {choice.content}
          </button>
        ))}
    </ul>
  );
};
export default QuestionComponent;
