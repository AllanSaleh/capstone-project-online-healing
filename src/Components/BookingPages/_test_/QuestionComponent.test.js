import * as React from 'react';
import QuestionComponent from "../QuestionComponent";
import * as renderer from "react-test-renderer";

it('Question Componenet renders correctly', () => {
  const tree = renderer
    .create(<QuestionComponent/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});