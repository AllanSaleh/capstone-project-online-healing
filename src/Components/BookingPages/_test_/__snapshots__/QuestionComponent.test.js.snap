// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Componenet renders correctly 1`] = `<ul />`;
